drivers/modem/modem_cellular.c

```diff
@@ -293,7 +317,48 @@ static void modem_cellular_chat_on_cgmm(struct modem_chat *chat, char **argv, ui
 		return;
 	}
 
-	strncpy(data->hwinfo, argv[1], sizeof(data->hwinfo) - 1);
+	strncpy(data->fw_version, argv[1], sizeof(data->fw_version) - 1);
+}
+
+static void modem_cellular_chat_on_csq(struct modem_chat *chat, char **argv, uint16_t argc,
+				       void *user_data)
+{
+	/* AT+CSQ returns a response +CSQ: <rssi>,<ber> where:
+	 * - rssi is a integer from 0 to 31 whose values describes a signal strength
+	 *   between -113 dBm for 0 and -51dbM for 31 or unknown for 99
+	 * - ber is an integer from 0 to 7 that describes the error rate, it can also
+	 *   be 99 for an unknown error rate
+	 */
+	struct modem_cellular_data *data = (struct modem_cellular_data *)user_data;
+
+	if (argc != 3) {
+		return;
+	}
+
+	/* Read rssi */
+	uint8_t rssi = atoi(argv[1]);
+
+	if (rssi == 99) {
+		return;
+	}
+
+	data->rssi = (-113 + (2 * rssi));
+}
+
+static void modem_cellular_chat_on_imsi(struct modem_chat *chat, char **argv, uint16_t argc,
+					void *user_data)
+{
+	struct modem_cellular_data *data = (struct modem_cellular_data *)user_data;
+
+	strncpy(data->imsi, (char *)argv[1], sizeof(data->imsi));
+}
+
+static void modem_cellular_chat_on_iccid(struct modem_chat *chat, char **argv, uint16_t argc,
+					 void *user_data)
+{
+	struct modem_cellular_data *data = (struct modem_cellular_data *)user_data;
+
+	strncpy(data->iccid, (char *)argv[1], sizeof(data->iccid));
```

Comment:
My modem (BG95) actually returns here "+QCCID: nnnnnnnnnnnnnnnnnnnn" for some reason, instead of directly outputting the ICCID. 

So after the last line, we can do something like this to get rid of the initial "+QCCID: ", taken from the `gsm_ppp.c` driver [here](https://github.com/zephyrproject-rtos/zephyr/blob/742de21eef248c52a24ea37495edbbc4e37c43ea/drivers/modem/gsm_ppp.c#L334).

```
if (data->sim_iccid[0] == '+') {
	/* Seen on U-blox SARA and Quectel BG95: 	.
	 * Skip over the +CCID bit, which other modems omit.
	 */
	char *p = strchr(data->sim_iccid, ' ');

	if (p) {
		size_t iccid_len = strlen(p+1);

		(void)memmove(data->sim_iccid, p+1, iccid_len+1);
	}
}
```